creator-node/src/services/sync/secondarySyncFromPrimaryUtils.ts

```diff
@@ -0,0 +1,112 @@
+import type {
+  ForceResyncAuthParams,
+  ForceResyncConfig,
+  ForceResyncSigningData,
+  SyncRequestAxiosData
+} from '../stateMachineManager/stateReconciliation/types'
+
+const _ = require('lodash')
+
+const { logger: genericLogger } = require('../../logging')
+const config = require('../../config')
+const ContentNodeInfoManager = require('../stateMachineManager/ContentNodeInfoManager')
+const {
+  recoverWallet,
+  generateTimestampAndSignature
+} = require('../../apiSigning')
+
+const generateDataForSignatureRecovery = (body: SyncRequestAxiosData) => {
```

Comment:
@vicky-g can we use apiSigning.js for this stuff?

Reply:
this does use apiSigning inherently, just a wrapper around this functionality. I want to not use the vanilla apiSigning because since we have scattered usage of calling /sync, I thought it would be good to have a file and function to specify all that

Reply:
+1 to dheeraj's point here

i dont think we need `signSyncData` or `recoverWalletfromSyncData`, since they are both just  one liners calling apiSigning functions which are appropriately abstracted

Reply:
my intention here is to remove the onus that the engineer reviewing the code should be the one knowing the api signatures / knowing what to pass. we should delegate responsibility to abstract functions for consistency

Reply:
i kinda feel the opposite here. i think this is actually more confusing for the reviewer since adds an additional layer of indirection and now we break the pattern of how we do this in other places. plus long term is it really feasible to have apiSigning, custom wrapper and the consumer consumes the wrapper? seems very indirect

Reply:
I think we should have actually abstracted out any usages on apiSigning to wrapper fns since the beginning. again, the engineers working on anything related to api signing should not need to know what the actual data needs to be for signatures and recovery, and instead focus on how it's used throughout the system. this reduces mental mapping. 

if you guys feel strongly I can change it but I dont agree with both of you guys on using vanilla apiSigning

Reply:
sounds like we should make apiSigning more approachable. i feel strongly that we should not have individual wrappers for stuff for example

sync -> sync utils -> apiSigning
other feature -> other feature utils -> apiSigning
...

creates so much more overhead. we can chat with the team about how much abstraction we would like but this seems like too much

Reply:
making wrapper functions is _not_ to make api signing more approachable. the whole intention of wrapper functions is to abstract out the details of what needs to be passed in. if you ask anyone right now what gets signed/recovered for verifying sp identity, they would have to look at the code and trace through the implementation, rather than read a helper function that specifies what is necessary for verification. abstraction helps reduce overhead for future engineers to need to understand data inputs/outputs. 

I dont think there really needs to be a discussion but im just saying it abstraction removes the onus of the engineer to understand particulates of inputs/outputs/flows by calling helper fns instead of writing vanilla code

Reply:
for now lets agree on a resolution and discuss this later in detail, i think this is a broader topic and we should pick an approach and stick to it. i would advocate changing it now but i don't want that to block merging if everything else is good to go

Reply:
im gonna change the wrapper functions to call the vanilla recover and signing, but im going to keep the function that generates the data to sign for api signatures

Reply:
that sounds good 👍

Reply:
nice, this is reasonable - thanks for updating